fix: Add types field to package.json exports

lachlancollins opened the pull request:
Hi @ConsoleTVs , this is a small PR that improves typescript compatibility. At the moment, [publint](https://publint.dev/sswr@2.0.0) and [ATTW](https://arethetypeswrong.github.io/?p=sswr%402.0.0) give a warning about the missing `exports["."]["types"]` field.

What's changed:
- Added `exports["."]["types"]` field
- Marked the package as `"type": "module"` (`.js` interpreted as ESM)
- Added `src` directory to output (probably not that useful since there aren't sourcemaps but still nice to be able to inspect the source)

lachlancollins commented:
Hi @ConsoleTVs, just wondering if you could check out this PR? Thanks!

ConsoleTVs commented:
Hi @lachlancollins, Would it be possible to update to Vite 5? I feel this change would also be welcome!

lachlancollins commented:
> Hi @lachlancollins, Would it be possible to update to Vite 5? I feel this change would also be welcome!

No worries! Since there were 2 lockfiles, I've removed `package-lock.json` in favour of `yarn.lock`, and updated Vite and TS. I've put the Svelte v5 alpha in peerDeps.
